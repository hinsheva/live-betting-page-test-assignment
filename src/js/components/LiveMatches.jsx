// @flow

import React from 'react';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Slider from 'react-slick';
import SideBlock from './SideBlock';
import Match from './Match';
import Error from './Error';
import Loading from './Loading';
import { mappedLiveEventsData, carouselStyle, getLiveEventsFromCache } from '../utils/utils';
import { STATES } from '../constants';

const LiveMatches = ({ current, send }: { current: string, send: Function }) => {
  const { liveEvents, error } = current.context;
  const { LOADING, LOADED, FAILED } = STATES;

  getLiveEventsFromCache({ send });

  return (
    <div id="content">
      <article>
        <h1>Live matches</h1>
        <p className="preamble">Here is a list of matches that are live right now.</p>
        <SideBlock />
        <div id="live-matches">
          <Slider {...carouselStyle}>
            {current.matches(LOADING) && <Loading />}
            {current.matches(LOADED) &&
              mappedLiveEventsData(liveEvents).map((match) => (
                <Match key={match.id} liveEvent={match} send={send} />
              ))}
            {current.matches(FAILED) && <Error error={error} />}
          </Slider>
        </div>
      </article>
    </div>
  );
};

export default LiveMatches;
