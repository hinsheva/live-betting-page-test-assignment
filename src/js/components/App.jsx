import React from 'react';
import '../../css/main.css';
import Header from './Header';
import Footer from './Footer';
import NavBar from './NavBar';
import LiveMatches from './LiveMatches';

const App = () => {
  return (
    <>
      <div id="container">
        <Header />
        <NavBar />
        <LiveMatches />
        <Footer />
      </div>
    </>
  );
};

export default App;
