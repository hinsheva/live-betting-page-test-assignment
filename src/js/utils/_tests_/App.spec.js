import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../../components/App';
import LiveMatches from '../../components/LiveMatches';

describe('App', () => {
  const props = {
    current: {
      context: {
        liveEvents: [],
      },
      matches: Function,
    },
    send: Function,
  };
  const renderer = TestRenderer.create(<App {...props} />);

  it('should render LiveMatches component with children', () => {
    const renderedTree = renderer.toJSON();
    expect(renderedTree.children.length).toBe(4);
  });

  it('should render LiveMatches component with children', () => {
    const renderedTree = renderer.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('renders correctly child component LiveMatches', () => {
    const childRenderer = new ShallowRenderer();
    childRenderer.render(<LiveMatches {...props} />);
    const result = childRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
