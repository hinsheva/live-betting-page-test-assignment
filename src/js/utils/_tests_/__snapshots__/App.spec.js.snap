// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly child component LiveMatches 1`] = `
<div
  id="content"
>
  <article>
    <h1>
      Live matches
    </h1>
    <p
      className="preamble"
    >
      Here is a list of matches that are live right now.
    </p>
    <SideBlock />
    <div
      id="live-matches"
    >
      <Slider
        autoplay={true}
        autoplaySpeed={3000}
        dots={false}
        infinite={true}
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
      >
        <Loading />
        <Error />
      </Slider>
    </div>
  </article>
</div>
`;

exports[`App should render LiveMatches component with children 1`] = `
<div
  id="container"
>
  <header>
    <div
      id="logo"
    >
      Unibet
    </div>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="instructions/index.html"
        >
          Navbar
        </a>
      </li>
    </ul>
  </nav>
  <div
    id="content"
  >
    <article>
      <h1>
        Live matches
      </h1>
      <p
        className="preamble"
      >
        Here is a list of matches that are live right now.
      </p>
      <aside>
        <h2>
          Live betting
        </h2>
        <p>
          Place your bets as the action unfolds. We offer a wide selection of live betting events and you can place both single and combination bets.
        </p>
        <p>
          You will be able to see an in-play scoreboard with the current result and match stats, while on selected events you will also be able to watch the action live with Unibet TV on the desktop site.
        </p>
      </aside>
      <div
        id="live-matches"
      >
        <div
          className="slick-slider slick-initialized"
        >
          
          <div
            className="slick-list"
          >
            <div
              className="slick-track"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                  "WebkitTransition": "",
                  "msTransform": "translateX(0px)",
                  "opacity": 1,
                  "transform": "translate3d(0px, 0px, 0px)",
                  "transition": "",
                }
              }
            >
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-current"
                data-index={0}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="loading"
                  >
                    <div>
                      Loading...
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          
          
        </div>
      </div>
    </article>
  </div>
  <footer>
    <div
      className="inner"
    >
      <p>
        Â© 1997-2015, Unibet. All rights reserved.
      </p>
    </div>
  </footer>
</div>
`;
